File: javascript/0007-rev-int.js
// ! Day2 // 9.29

var reverse = function(x) {
  let negative = x < 0;
  let range = 2 ** 31 - 1;
  let reversed = 0;

  if (negative) {
    x = x * -1;
  }

  while (x > 0) {
    reversed = reversed * 10 + (x % 10);
    x = Math.floor(x / 10);
  }

  if (reversed > range) {
    return 0;
  }

  return negative ? reversed * -1 : reversed;
};

File: ruby/0007-rev-int.rb
def reverse(x)
   
    if x < 0
      negative = true
    end
  
    x = x.abs
    reversed_integer = 0
  
    while x > 0
      reversed_integer = reversed_integer * 10 + (x % 10)
      x /= 10
        
    end
  
  
      
      if reversed_integer > 2 ** 31
          return 0
      end
  
    negative ? reversed_integer * -1 : reversed_integer
  end